### ISSUE TITLE
Update to Java 11 (?)

### ISSUE THREAD

**Issue by MartinWahnschaffe** [hzi-braunschweig/SORMAS-Project#6699]
### Feature Description
Once #6686 is done we finally have Java 8 support for the SORMAS API project. 

### Problem Description
With Java 11 there is a newer LTS version available that brings some benefits (see https://medium.com/testvagrant/migrate-java-8-to-java-11-12-9ab082f335e8 - the Oracle part is not relevant for us, since we are relying on OpenJDK anyway).

With the upgrade to Android Gradle Plugin 7+, we now also have support for Java 11 in Android:
https://developer.android.com/studio/releases/gradle-plugin#java-11

This roadmap provided by azul (zulu) is a bit confusing, though: https://www.azul.com/products/azul-support-roadmap/
It says the support for Java 8 will be longer for Java 11.

### Proposed Change

- [ ] clarify whether the azul roadmap is a reason to stay with Java 8
- [ ] update the used java version of all projects to 11
- [ ] test Java 11 for Android. I had a try with it and was able to use Java 11 API (String.isBlank) in an SORMAS API method that was then used by the Android app and ran on a version 24 emulator. I was not able to directly use String.isBlank in Android, though. It should be clarified why this is the case. There is currently no documentation available for this - maybe the release of Android Studio 2021 bumblebee will clarify this.

**Comment by JonasCir** [comment]
@MartinWahnschaffe 
I don't think we should stick with zulu JDK in the first place. I use openjdk elusively since my very first day in the project and there was never an issue. For me, the zulu JDK feels like a shady rebrand of openjdk.

**Comment by JonasCir** [comment]
Also there is this issue, not sure how accurate it is now: https://github.com/hzi-braunschweig/SORMAS-Project/issues/2617

**Comment by MartinWahnschaffe** [comment]
Answer I received on why Java 8 LTS is supported longer than Java 11 LTS

> The incompatibility between Java 8 and Java 11 makes the migration of Java 8 applications much more difficult. In most cases, it makes more economic sense to let the application die a "natural death" on Java 8. This situation affects both enterprise applications and embedded solutions. We reflect this in the long support.
> 
> From Java 11 onwards, migration should be much easier. Applications should follow the new Java roadmap faster and therefore a long support for Java 11 should not be necessary. 

From my point of view this is clearly no reason to stay with Java 8, instead it's a good reason to go with Java 11 and have an easy migration to newer LTS versions once they are available.

**Comment by StefanKock** [comment]
> @MartinWahnschaffe I don't think we should stick with zulu JDK in the first place. I use openjdk elusively since my very first day in the project and there was never an issue. For me, the zulu JDK feels like a shady rebrand of openjdk.

Decided in todays planning with @MartinWahnschaffe: It does not matter which open available JDK to use, but zulu feels to be more caring, better download pages and was never reported to be a problem. Conclusion: We stay with zulu

**Comment by MartinWahnschaffe** [comment]
Update to Java 11 was surprisingly easy. The update to Android API 33 (#11032) has added seemingly all (?) java API methods. See https://developer.android.com/sdk/api_diff/33/changes (I could not find an article clearly summarizing this).

To test this I have added a call to the Java 11 `String.isBlank` method in LoginActivity, which works fine. Doing the same in a helper method in sormas-api and then calling the helper method form sormas-app also works.

I have also tested backwards compatibility for Android, though with our current minimum SDK version of 26, this is not of much relevance. Setting the minimum SDK version to 1 directly shows method calls that are not supported in Android Studio and also when building.
If needed we could also use desugaring to increase backwards compatibility for calls to Java API methods: https://developer.android.com/studio/write/java11-default-support-table

I went a step further to made sure we will also be noticed when we are using something that is not supported. To do so I have set the sormas-api Java version to 17 and used the Java 17 `RandomGeneratorFactory `class in the helper method. Calling this method from Android leads to the following build error:

> Missing class java.util.random.RandomGeneratorFactory (referenced from: java.util.stream.IntStream de.symeda.sormas.api.utils.DataHelper.getPseudoInts(java.lang.String, int))

Conclusion: 
1. We could also update sormas-api to Java 17.
2. In case we are using any unsupported Java method in sormas-api in a method that is called from sormas-app, the builder will tell us, so we are not in risk of running into this at runtime.

Based on that I decided to completely remove the java version entry form the maven pom of sormas-api.

**Very important:** After pulling the changes, you need to use the "Maven > Reload Project" functionality of IntelliJ to update the Java version of the module.
